contributing/development/cpp_usage_guidelines.rst

```diff
@@ -53,18 +53,69 @@ as Godot provides its own data types (among other things).
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Vector``             | ``std::vector``          | **Use this as the "default" vector type.** Uses copy-on-write semantics.            |
+|                        |                          | This means it's generally slower but can be copied around almost for free.          |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``LocalVector``        | ``std::vector``          | Closer to ``std::vector`` in semantics. In most situations, ``Vector`` should be    |
+|                        |                          | preferred.                                                                          |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Array`` 📜           | ``std::vector``          | Values can be of any Variant type. No static typing is imposed.                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``TypedArray`` 📜      | ``std::vector``          | Equivalent to ``Array`` but with static typing for its elements.                    |
+|                        |                          | Not to be confused with ``Packed*Array``, which is internally a ``Vector``.         |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Packed*Array`` 📜    | ``std::vector``          | Equivalent to ``Vector``, e.g. ``PackedColorArray = Vector<Color>``.                |
+|                        |                          | Only a limited list of packed array types are available                             |
+|                        |                          | (use ``TypedArray`` otherwise).                                                     |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``Span``               | ``std::span``            | Represents read-only access to a contiguous array without needing to copy any data. |
+|                        |                          | See `pull request description <https://github.com/godotengine/godot/pull/100293>`__ |
+|                        |                          | for details.                                                                        |
++------------------------+--------------------------+-------------------------------------------------------------------------------------+
+| ``HashSet``            | ``std::set``             | **Use this as the "default" set type.**                                             |
```

Comment:
As both are hash containers, especially since `HashSet` doesn't offer any insertion ordering
```suggestion
| ``HashSet``            | ``std::unordered_set``   | **Use this as the "default" set type.**                                             |
```